include/boost/geometry/algorithms/detail/closest_points/default_strategies.hpp

```diff
@@ -0,0 +1,167 @@
+// Boost.Geometry
+
+// Copyright (c) 2019, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_ALGORITHMS_DETAIL_CLOSEST_POINTS_DEFAULT_STRATEGIES_HPP
+#define BOOST_GEOMETRY_ALGORITHMS_DETAIL_CLOSEST_POINTS_DEFAULT_STRATEGIES_HPP
+
+#include <boost/geometry/core/cs.hpp>
+#include <boost/geometry/core/tag.hpp>
+#include <boost/geometry/core/tag_cast.hpp>
+#include <boost/geometry/core/tags.hpp>
+#include <boost/geometry/core/point_type.hpp>
+#include <boost/geometry/core/reverse_dispatch.hpp>
+
+#include <boost/geometry/strategies/closest_points.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail { namespace closest_points
+{
+
+
+
+// Helper metafunction for default strategy retrieval
+template
+<
+    typename Geometry1,
+    typename Geometry2 = Geometry1,
+    typename Tag1 = typename tag_cast
+        <
+            typename tag<Geometry1>::type, pointlike_tag
+        >::type,
+    typename Tag2 = typename tag_cast
+        <
+            typename tag<Geometry2>::type, pointlike_tag
+        >::type,
+    bool Reverse = geometry::reverse_dispatch<Geometry1, Geometry2>::type::value
+>
+struct default_strategy
```

Comment:
Shouldn't this file be in directory `strategies`?

Reply:
Ok, I now see that it corresponds to `algorithms/detail/distance/default_strategies.hpp`. So ignore.

Reply:
I followed the design of distance, see https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/algorithms/detail/distance/default_strategies.hpp